packages/@lwc/engine-core/src/framework/def.ts

```diff
@@ -52,6 +53,8 @@ export interface ComponentDef {
     propsConfig: Record<string, PropType>;
     methods: PropertyDescriptorMap;
     template: Template;
+    preferNativeShadow: boolean;
+    renderMode: 'light' | 'shadow';
```

Comment:
this means we will be doing string comparison in multiple places... can we instead use the typescript enum trick here? In typescript, the enum definition can have a runtime value (in this case numeric), so the code still uses the enums everywhere, but it is not a string comparison, something like that will eliminate the string comparison per instance that worries me.